lib/plugins/azurepush.js

```diff
@@ -0,0 +1,323 @@
+        // *****REMOVE*****: Beta logging
+        if (err) {
+          logTrelloIssue(env, '0x08-Failed to verify index on alert in mongodb');
+        }
+        // END: *****REMOVE*****
+        
+        // TODO: Handle error
+        console.log(err, doc);
+      });
+      ctx.store.collection(env.azurepush_collection).ensureIndex('settings.info', function indexAnnouncement(err, doc) {
+        // *****REMOVE*****: Beta logging
+        if (err) {
+          logTrelloIssue(env, '0x09-Failed to verify index on info in mongodb');
+        }
+        // END: *****REMOVE*****
+        
+        // TODO: Handle error
+        console.log(err, doc);
+      });
+      
+      // More info on request body: https://msdn.microsoft.com/en-us/library/mt621153.aspx
+      var installation = {
+        installationId: request.installationId
+        , pushChannel: request.deviceToken
+        , templates: {
+          nightscout: {
+            tags: tags
+          }
+        }
+      };
+
+      // 'platform': https://developer.xamarin.com/guides/xamarin-forms/platform-features/device/#Device.OS
+      switch (request.platform) {
+        case 'iOS':
+          installation.platform = 'apns';
+          installation.templates.nightscout.body = '{\"aps\":{\"alert\":\"$(message)\",\"sound\":\"$(sound)\",\"category\":\"$(category)\"},\"eventName\":\"$(eventName)\",\"group\":\"$(group)\",\"key\":\"$(key)\",\"level\":\"$(level)\",\"title\":\"$(title)\"}';
+          break;
+        case 'Android':
+          installation.platform = 'gcm';
+          installation.templates.nightscout.body = '{\"data\":{\"message\":\"$(message)\",\"eventName\":\"$(eventName)\",\"group\":\"$(group)\",\"key\":\"$(key)\",\"level\":\"$(level)\",\"sound\":\"$(sound)\",\"title\":\"$(title)\"}}';
+          break;
+        case 'WinPhone': //Not yet supported
+        case 'Windows': //Not yet supported
+        default: //No platform specified, or invalid platform
+          // *****REMOVE*****: Beta logging
+          logTrelloIssue(env, '0x10-Register received without supported platform');
+          // END: *****REMOVE*****
+
+          return callback({message: 'Platform not supported'});
+      }
+
+      notificationHubService.createOrUpdateInstallation(installation, function (error, response) {
+        // *****REMOVE*****: Beta logging
+        if (error) {
+          logTrelloIssue(env, '0x11-Failed to create or update installation');
+        }
+        // END: *****REMOVE*****
+
+        callback(error, {installationId: installation.installationId});
+      });
+    }
+    else {
+      // *****REMOVE*****: Beta logging
+      logTrelloIssue(env, '0x12-Register received without platform');
+      // END: *****REMOVE*****
+
+      // No platform, required parameter
+      return callback({message: 'Platform required'});
+    }
+  };
+
+  azurepush.ack = function ack(request, callback) {
+    if (request && request.level) {
```

Comment:
Which extra collection are you talking about?  pushclients?  Or the notifications collection?

And let me see if I understand: instead of registering the device for push notifications generically, we would register it specific to a subject.  Then, that subject either has global ack or local ack.  Global would snooze for everyone, local would only prevent further notifications for that specific device and subject's site?